File: Golang/passwordcracker.go
package Golang

import (
	"fmt"
	"time"
)

func bruteforce(pw string) (int, string) {
	// attempts := 0
	// chars := strings.Split("0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ!\"#$%&'()*+,-./:;<=>?@[\\]^_`{|}~", "")
	// for i := 1; i <= len(pw); i++ {
	// 	for _, guess := range cartesian.Iter(chars, i) {
	// 		attempts++
	// 		if strings.Join(guess, "") == pw {
	// 			return attempts, strings.Join(guess, "")
	// 		}
	// 	}
	// }
	// return attempts, ""
}

func main() {
	startTime := time.Now()
	var pw string
	fmt.Print("Input the password to crack: ")
	fmt.Scanln(&pw)
	attempts, guess := bruteforce(pw)
	if guess != "" {
		fmt.Printf("Password cracked in %d attempts. The password is %s.\n", attempts, guess)
	} else {
		fmt.Printf("Password not cracked after %d attempts.\n", attempts)
	}
	fmt.Printf("Time taken to complete execution: %s\n", time.Since(startTime))
}

File: Python/passwordcracker/passwordcracker.py
# Simple Brute Force Password Cracker
import string
from itertools import product
import time

def bruteforce(pw):
    attempts = 0
    chars = string.printable
    for i in range(1, len(pw)+1):
        # giving cartesian product
        for guess in product(chars, repeat=i):
            attempts += 1
            guess = ''.join(guess)
            if guess == pw:
                return (attempts, guess)
    return (attempts, None)

start_time = time.time()
pw = input("Input the password to crack: ")
attempts, guess = bruteforce(pw)
print(f"Password cracked in {attempts} attempts. The password is {guess}.") if guess else print(f"Password not cracked after {attempts} attempts.")
print(f"Time taken to complete execution: {time.time() - start_time}")